samples/DotNet/Azure.Messaging.EventHubs/AnomalyDetector/Producer/Program.cs

```diff
@@ -0,0 +1,106 @@
+        private const string EventHubName = "<EVENT HUB NAME>";
+        private const string TransactionsDumpFile = "mocktransactions.csv";
+
+        private static EventHubProducerClient producerClient;
+
+        public static int Main()
+        {
+            return MainAsync().GetAwaiter().GetResult();
+        }
+
+        private static async Task<int> MainAsync()
+        {
+            // create an Event Hubs Producer client using the namespace connection string and the event hub name
+            producerClient = new EventHubProducerClient(EventHubConnectionString, EventHubName);
+
+            // send messages to the event hub
+            await SendMessagesToEventHubAsync(1000);
+
+            await producerClient.CloseAsync();
+
+            Console.WriteLine("Press [enter] to exit.");
+            Console.ReadLine();
+
+            return 0;
+        }
+
+        // Creates an Event Hub client and sends messages to the event hub.
+        private static async Task SendMessagesToEventHubAsync(int numMessagesToSend)
+        {
+            var eg = new EventGenerator();
+
+            IEnumerable<Transaction> transactions = eg.GenerateEvents(numMessagesToSend);
+            
+            if (File.Exists(TransactionsDumpFile))
+            {
+                // exceptions not handled for brevity
+                File.Delete(TransactionsDumpFile);
+            }
+
+            File.AppendAllText(
+                TransactionsDumpFile, 
+                $"CreditCardId,Timestamp,Location,Amount,Type{Environment.NewLine}");
+
+            foreach (var t in transactions)
+            {
+                try
+                {
+                    // we don't send the transaction type as part of the message.
+                    // that is up to the downstream analytics to figure out!
+                    // we just pretty print them here so they can easily be compared with the downstream
+                    // analytics results.
+                    var message = t.Data.ToJson(); 
+
+                    if (t.Type == TransactionType.Suspect)
+                    {
+                        var fc = Console.ForegroundColor;
+                        Console.ForegroundColor = ConsoleColor.Yellow;
+
+                        Console.WriteLine($"Suspect transaction: {message}");
+
+                        Console.ForegroundColor = fc; // reset to original
+                    }
+                    else
+                    {
+                        Console.WriteLine($"Regular transaction: {message}");
+                    }
+
+                    var line = $"{t.Data.CreditCardId},{t.Data.Timestamp.ToString("o")},{t.Data.Location},{t.Data.Amount},{t.Type}{Environment.NewLine}";
+
+                    File.AppendAllText(TransactionsDumpFile, line);
+
+                    // prepare a batch of events to send to the event hub. only one event in this case. 
+                    EventDataBatch eventBatch = await producerClient.CreateBatchAsync();
```

Comment:
I'd recommend against this pattern as a best practice.   

Ideally, you'd build the batch until it cannot hold more events before sending, rather than doing single event batches.   In case it is helpful, there's a [sample](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample04_PublishMultipleEventBatches.cs#L80) available to illustrate.   _(lines 80-97 are the relevant part)_

Reply:
Done. Adding all the messages to the batch and then send the batch at once to the event hub.